### PR TITLE
C sharp linq

### PR DESCRIPTION
**Pull request by KarinaKrinitskaya**

### CODE REVIEW

File: CSharpLinq/CSharpLinq/Program.cs [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+PrintCollection(sortAnimal);
+
+#endregion
+
+#region Task_4
+
+List<dynamic> cities = new List<dynamic>() { "Minsk", "Warsaw", "Lisbon", "Porto", "Barselona", "Rome" };
+
+var resultCities = cities.Where(city => city[0].Equals('P') && city[city.Length - 1].Equals('o'));
+
```
Comment (dnnyhlwdplc) [inline CSharpLinq/CSharpLinq/Program.cs:54]: Try StartsWith, EndsWith :)

File: CSharpLinq/CSharpLinq/Program.cs [hunk 2]
```diff
@@ -0,0 +1,186 @@
+﻿using CSharpOOP2;
+using System;
+using System.Collections.Generic;
+//using System.Text.RegularExpressions;
+//using static System.Net.Mime.MediaTypeNames;
+
+#region Task_1
+
+List<dynamic> numbers = new List<dynamic>() { 545, 4, 2, 32, 8, 99, 7, 5, 7, 9 };
```
Comment (dnnyhlwdplc) [inline CSharpLinq/CSharpLinq/Program.cs:9]: Why dynamic and not object? :)

File: CSharpLinq/CSharpLinq/Program.cs [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+
+//Lacrimosa dies illa Qua resurget ex favilla Judicandus homo reus
+
+var someStr = Console.ReadLine();
+
+//EXPERIMENT
+//regular Expressions for //"LacrimosaDiesIlla"
+//string[] split = Regex.Split(someStr, @"(?<!^)(?=[A-Z])");
+
+var resultString = someStr.Split(' ').Where(s => char.IsUpper(s[0]));
```
Comment (dnnyhlwdplc) [inline CSharpLinq/CSharpLinq/Program.cs]: Hmm, you should have printed words consist of uppercase letters only :)

File: CSharpLinq/CSharpLinq/Program.cs [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+
+PrintCollection(resultString);
+
+#endregion
+
+#region Task_6
+
+List<dynamic> collections4 = new List<dynamic>() { 7, 9, 11, 15, 13, 9, 9, 12, 14, 9 };
+
+var result4 = collections4.Distinct().OrderByDescending(c => c);
```
Comment (dnnyhlwdplc) [inline CSharpLinq/CSharpLinq/Program.cs]: In task you should have used Reverse() :)

File: CSharpLinq/CSharpLinq/Program.cs [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+
+#endregion
+
+#region Task_7
+
+List<int> collections5 = new List<int>() { 100, 7, 110, 0, 15, 90, 87, 33, 1, 54 };
+
+var someNumber = ReadValue<int>(int.TryParse);
+
+bool has = collections5.Contains(someNumber);
```
Comment (dnnyhlwdplc) [inline CSharpLinq/CSharpLinq/Program.cs:101]: Interesting! Contsins is also suitable here :)

File: CSharpLinq/CSharpLinq/Program.cs [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+{
+    foreach (dynamic n in someList)
+    {
+        Console.WriteLine(n);
+    }
+}
+
+Console.ReadLine();
+
+static T ReadValue<T>(ParseDelegate<T> parser)
```
Comment (dnnyhlwdplc) [inline CSharpLinq/CSharpLinq/Program.cs:170]: I suppose you can skip next 2 lessons :D

File: CSharpLinq/CSharpLinq/Program.cs [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+    Console.WriteLine("The collection hasn't a number");
+}
+
+#endregion
+
+#region Task_8
+
+List<string> collections6 = new List<string>() { "Pupum", "Lalala", "Patapon", "Chaka-Chaka" };
+
+int sumSymbols = collections6.Sum(s => s.Length);
```
Comment (dnnyhlwdplc) [inline CSharpLinq/CSharpLinq/Program.cs:118]: Perfect!

**Review by dnnyhlwdplc — APPROVED** [review]

File: CSharpLinq/CSharpLinq/Program.cs (hunk 1 above)
Comment (KarinaKrinitskaya) [reply CSharpLinq/CSharpLinq/Program.cs:54]: Tried:)

File: CSharpLinq/CSharpLinq/Program.cs (hunk 2 above)
Comment (KarinaKrinitskaya) [reply CSharpLinq/CSharpLinq/Program.cs:9]: Answered on skype:)

File: CSharpLinq/CSharpLinq/Program.cs (hunk 3 above)
Comment (KarinaKrinitskaya) [reply CSharpLinq/CSharpLinq/Program.cs]: Fixed:)

File: CSharpLinq/CSharpLinq/Program.cs (hunk 4 above)
Comment (KarinaKrinitskaya) [reply CSharpLinq/CSharpLinq/Program.cs]: Fixed:)

**Pull request merged**